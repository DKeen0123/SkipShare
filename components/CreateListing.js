import React, { Component } from 'react'
import { Text, View, Button } from 'react-native'
import { Form, Item, Input, Label } from 'native-base';

export default class CreateListing extends Component {
  state = {
    picture: '',
    description: '',
    address: '',
    contactNumber: ''
  }

  submitForm = () => {}

  render() {
    return (
      <View>
        <Text> Create a new listing </Text>
         <Form>
        <Item floatingLabel>
          <Label>Picture</Label>
          <Input
            value={this.state.picture}
            onChangeText={picture => this.setState({ picture })}
          />
        </Item>

        <Item floatingLabel>
          <Label>Description</Label>
          <Input
            value={this.state.description}
            onChangeText={description => this.setState({ description })}
          />
        </Item>

        <Item floatingLabel>
          <Label>Address</Label>
          <Input
            value={this.state.address}
            onChangeText={address => this.setState({ address })}
          />
        </Item>

        <Item floatingLabel>
          <Label>Contact no.</Label>
          <Input
            value={this.state.contactNumber}
            onChangeText={contactNumber => this.setState({ contactNumber })}
          />
        </Item>
        <Button title='Create your listing!' onPress={this.submitForm} />
      </Form>
      </View>
    )
  }
}