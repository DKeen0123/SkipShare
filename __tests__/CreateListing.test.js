import React from 'react';
import { shallow } from 'enzyme';
import CreateListing from '../components/CreateListing';

const createListing = shallow(<CreateListing />);

it('renders without crashing', () => {
  expect(createListing).toMatchSnapshot();
});