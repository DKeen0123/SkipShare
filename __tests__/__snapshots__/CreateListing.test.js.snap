// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Create a new listing 
  </Text>
</Component>
`;
